Update workflow setup script for free-forecast ensemble

XianwuXue-NOAA opened the pull request:
<!-- PLEASE READ -->
<!--
Before opening a PR, please note these guidelines:

- Each PR should only address ONE topic and have an associated issue
- No hardcoded or paths to personal directories should be present
- No temporary or backup files should be committed
- Any code that was disabled by being commented out should be removed
-->

**Description**

As we move towards the merger of GEFS into the GFS system, this PR is to add new feature which to generate workflows with an arbitrary number of free-forecast ensemble members in addition to the number of GEFS members.

<!-- Please include relevant motivation and context. -->
**Please include a summary of the change and which issue is fixed.**

 Improve ush/rocoto/setup_workflow_fcstonly.py to add some metatasks (getic, init and fcst) for GEFS to xml file.

<!-- List any dependencies that are required for this change. -->

<!-- Please provide reference to the issue this pull request is addressing. -->
<!-- For e.g. Fixes #IssueNumber -->

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

It has been tested on Hera.

<!-- Please describe the tests that you ran to verify your changes and on the platforms these tests were conducted. -->
<!-- Provide instructions so we can reproduce. -->
<!-- Please also list any relevant details for your test configuration -->

<!-- Use the following as a guide to list your tests and delete options that are not relevant. Expand as necessary. -->
<!--
- [ ] Clone and Build tests on WCOSS Dell P3
- [ ] Cycled test on Orion
- [ ] Forecast-only test on Hera
-->
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] Any dependent changes have been merged and published

Dep: #849

XianwuXue-NOAA committed:
Update workflow setup script for free-forecast ensemble

Add getic, init and fcst metatasks

 On branch feature/gefs_v13_828_add_ws
	modified:   ush/rocoto/setup_workflow_fcstonly.py

Refs: #828

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_828_add_ws

WalterKolczynski-NOAA commented on ush/rocoto/setup_workflow_fcstonly.py:
```diff
@@ -207,6 +219,13 @@ def get_postgroups(post, cdump='gdas'):
         fhout_hf = post['FHOUT_HF_GFS']
         fhrs_hf = list(range(fhmin, fhmax_hf + fhout_hf, fhout_hf))
         fhrs = fhrs_hf + list(range(fhrs_hf[-1] + fhout, fhmax + fhout, fhout))
+    elif cdump in ['gefs']:
```
This looks to be duplicate of 'gfs'. Maybe just add 'gefs' to that list unless/until we create a separate set of variables?

XianwuXue-NOAA replied:
Currently, I am not sure whether GEFS will need other variables, and gdas also has a block of statements, so I 'd like to keep this block and improve them in the future.

WalterKolczynski-NOAA commented on ush/rocoto/setup_workflow_fcstonly.py:
```diff
@@ -299,40 +320,68 @@ def get_workflow(dict_configs, cdump='gdas'):
         if hpssarch in ['YES']:
             deps = []
-            data = '&ROTDIR;/&CDUMP;.@Y@m@d/@H/atmos/INPUT/sfc_data.tile6.nc'
+            if cdump == "gefs":
```
To match the normal python idiom (and the rest of the code), these should all be `if cdump in ['gefs'].

WalterKolczynski-NOAA commented on ush/rocoto/setup_workflow_fcstonly.py:
```diff
@@ -458,8 +521,11 @@ def get_workflow(dict_configs, cdump='gdas'):
                               metatask='post', varname=varname1, varval=varval1, vardict=vardict)
-    tasks.append(task)
-    tasks.append('\n')
+    if cdump == "gefs":
```
Add a comment to these noting they are temporary bypasses for GEFS until we want to add these tasks to the GEFS task mesh.

WalterKolczynski-NOAA commented:
Like the other PR, this one needs to have its title and description filled out properly.

XianwuXue-NOAA committed:
Add some comments to explain some lines

 On branch feature/gefs_v13_828_add_ws
	modified:   ush/rocoto/setup_workflow_fcstonly.py

Refs: #828

XianwuXue-NOAA committed:
Change the == to in to match the normal python idiom
 On branch feature/gefs_v13_828_add_ws
	modified:   ush/rocoto/setup_workflow_fcstonly.py

Refs: #828

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA I just push two commits to make the changes based on your comments. For the rest comment, I'd like to keep the block for the future's improvement.

aerorahul reviewed (commented):
I would like to review this in a little more detail.
Please hold from merging.

aerorahul commented:
@XianwuXue-NOAA 
Please provide a description of what the changes are and what they intend to accomplish.

XianwuXue-NOAA commented:
@aerorahul I changed ush/rocoto/setup_workflow_fcstonly.py to add some metatasks (getic, init and fcst) for GEFS to xml file.

WalterKolczynski-NOAA commented:
> @aerorahul I changed ush/rocoto/setup_workflow_fcstonly.py to add some metatasks (getic, init and fcst) for GEFS to xml file.

You need a more complete description, and it needs to be in the first post, just like a regular commit message. It should describe what and why. And a relevant, less cryptic, title.

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA Thanks a lot. I will modify the first post.

XianwuXue-NOAA committed:
Fix a typo

 On branch feature/gefs_v13_828_add_ws
	modified:   ush/rocoto/setup_workflow_fcstonly.py

Refs: #828

XianwuXue-NOAA committed:
Change NMEM_ENKF to NMEM_GEFS for GEFS

 On branch feature/gefs_v13_828_add_ws
	modified:   ush/rocoto/setup_workflow_fcstonly.py

Refs: #828

XianwuXue-NOAA committed:
Change NMEM_GEFS back to NMEM_ENKF

 On branch feature/gefs_v13_828_add_ws
	modified:   ush/rocoto/setup_workflow_fcstonly.py

Refs: #828

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_828_add_ws

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_828_add_ws

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_828_add_ws

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_828_add_ws

 Conflicts:
	ush/rocoto/setup_workflow_fcstonly.py (delete)

 On branch feature/gefs_v13_828_add_ws

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_828_add_ws

XianwuXue-NOAA committed:
Improve to add GEFS workflow

 On branch feature/gefs_v13_828_add_ws
	modified:   env/HERA.env
	modified:   env/ORION.env
	modified:   parm/config/config.resources
	modified:   workflow/applications.py
	modified:   workflow/rocoto/workflow_tasks.py
	modified:   workflow/rocoto/workflow_xml.py

Refs: #828

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_828_add_ws
